### PR TITLE
Add support for token authentication with network proxy

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add support for token based authentication for konnectivity server to authenticate konnectivity agent.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

`KUBE_ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE=true ./cluster/kube-up.sh` to set up

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/assign @caesarxuchao 
/cc @cheftako 
/cc @dberkov
/sig api-machinery

### CODE REVIEW

File: cluster/gce/addons/konnectivity-agent/konnectivity-agent-ds.yaml [hunk 1]
```diff
@@ -28,14 +28,15 @@ spec:
           hostPath:
             path: /etc/srv/kubernetes/pki/konnectivity-agent
       containers:
-        - image: us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-agent:v0.0.4
+        - image: gcr.io/jying-gke-dev/proxy-agent-amd64:d3891c1adc6708e767ef4da54df1837fed956609
```
Comment (Jefftree) [inline cluster/gce/addons/konnectivity-agent/konnectivity-agent-ds.yaml]: Will need to be updated with a new release

File: cluster/gce/addons/konnectivity-agent/konnectivity-agent-rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: konnectivity-agent
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+---
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (dberkov) [inline cluster/gce/addons/konnectivity-agent/konnectivity-agent-rbac.yaml]: Is it a right place to create a server binding permissions on agent file?

File: cluster/gce/manifests/konnectivity-server.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ spec:
   hostNetwork: true
   containers:
   - name: konnectivity-server-container
-    image: us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-server:v0.0.4
+    image: gcr.io/jying-gke-dev/proxy-server-amd64:d3891c1adc6708e767ef4da54df1837fed956609
```
Comment (dberkov) [inline cluster/gce/manifests/konnectivity-server.yaml]: same here, you need to update with a new release

File: cluster/gce/addons/konnectivity-agent/konnectivity-agent-rbac.yaml (hunk 2 above)
Comment (Jefftree) [reply cluster/gce/addons/konnectivity-agent/konnectivity-agent-rbac.yaml]: Thanks for the catch. This should be bundled with the konnectivity server creation.

**Comment by caesarxuchao** [comment]
The configuration changes are consistent with https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/51.

/lgtm
/approve

File: cluster/gce/gci/configure-helper.sh [hunk 4]
```diff
@@ -652,8 +652,13 @@ function create-master-auth {
     append_or_replace_prefixed_line "${known_tokens_csv}" "${GCE_GLBC_TOKEN},"                "system:controller:glbc,uid:system:controller:glbc"
   fi
   if [[ -n "${ADDON_MANAGER_TOKEN:-}" ]]; then
-    append_or_replace_prefixed_line "${known_tokens_csv}" "${ADDON_MANAGER_TOKEN},"   "system:addon-manager,uid:system:addon-manager,system:masters"
+    append_or_replace_prefixed_line "${known_tokens_csv}" "${ADDON_MANAGER_TOKEN},"           "system:addon-manager,uid:system:addon-manager,system:masters"
```
Comment (lavalamp) [inline cluster/gce/gci/configure-helper.sh]: s/           / /

Comment (lavalamp) [reply cluster/gce/gci/configure-helper.sh]: (there should be 1 space instead of a bunch of spaces and a tab there)

Comment (lavalamp) [reply cluster/gce/gci/configure-helper.sh]: actually just revert this change?

File: cluster/gce/manifests/konnectivity-server.yaml [hunk 5]
```diff
@@ -51,5 +51,21 @@ spec:
       path: /etc/srv/kubernetes/pki
   - name: konnectivity-uds
     hostPath:
-      path: /etc/srv/kubernetes/konnectivity
+      path: /etc/srv/kubernetes/konnectivity-server
       type: DirectoryOrCreate
+---
```
Comment (lavalamp) [inline cluster/gce/manifests/konnectivity-server.yaml]: it's weird to have a second blob in this .yaml, when you made a separate file above?

File: cluster/gce/gci/configure-helper.sh (hunk 4 above)
Comment (Jefftree) [reply cluster/gce/gci/configure-helper.sh]: This fixes the alignment with the previous lines. Would you still like me to revert?

**Comment by lavalamp** [comment]
/approve

/hold

for nit fixes

File: cluster/gce/manifests/konnectivity-server.yaml (hunk 5 above)
Comment (Jefftree) [reply cluster/gce/manifests/konnectivity-server.yaml]: Each cluster addon has its own directory but all the manifests share the same directory (at least in gce). Didn't want to pollute the manifests directory.

File: cluster/gce/gci/configure-helper.sh (hunk 4 above)
Comment (lavalamp) [reply cluster/gce/gci/configure-helper.sh]: ack, sorry, the diff obscured this.

File: cluster/gce/manifests/konnectivity-server.yaml (hunk 5 above)
Comment (lavalamp) [reply cluster/gce/manifests/konnectivity-server.yaml]: I see, ok.

**Comment by Jefftree** [comment]
/hold 
(Double check network proxy build image first)

File: cluster/gce/addons/konnectivity-agent/konnectivity-agent-ds.yaml [hunk 6]
```diff
@@ -22,20 +22,16 @@ spec:
...
         - key: "CriticalAddonsOnly"
           operator: "Exists"
-      hostNetwork: true
-      volumes:
-        - name: pki
-          hostPath:
-            path: /etc/srv/kubernetes/pki/konnectivity-agent
       containers:
-        - image: us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-agent:v0.0.4
+        - image: us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-agent:v0.0.6
```
Comment (caesarxuchao) [inline cluster/gce/addons/konnectivity-agent/konnectivity-agent-ds.yaml]: this is still 0.0.6?

**Pull request merged**